### ISSUE TITLE
gracefully handle 404 on fetch

### ISSUE THREAD

**Issue by igalic** [iocage/libiocage#378]
```
# /usr/local/bin/ioc fetch --release 11.2-BETA3
[+] FetchRelease@11.2-BETA3: OK [85.719s]: 4
  [+] ReleasePrepareStorage@11.2-BETA3: OK [0.049s]: 1
  [+] ReleaseDownload@11.2-BETA3: OK [42.069s]: 1
  [+] ReleaseExtraction@11.2-BETA3: OK [43.601s]: 1
[+] ReleaseConfiguration@11.2-BETA3: OK [0.01s]: 1
[-] ReleaseUpdatePull@11.2-BETA3: FAILED [0.569s]: 1
```
```python-traceback
Traceback (most recent call last):
  File "/usr/local/bin/ioc", line 10, in <module>
    sys.dd:exit(cli())
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/iocage/cli/fetch.py", line 139, in cli
    fetch_updates=fetch_updates
  File "/usr/local/lib/python3.6/site-packages/iocage/cli/__init__.py", line 71, in print_events
    for event in generator:
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/Release.py", line 569, in fetch
    for event in self.updater.fetch():
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ResourceUpdater.py", line 255, in fetch
    self._pull_updater()
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ResourceUpdater.py", line 225, in _pull_updater
    local=f"{self.host_updates_dir}/{self.update_script_name}"
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ResourceUpdater.py", line 207, in _download_updater_asset
    _request.urlretrieve(url, local)  # nosec: url validated
  File "/usr/local/lib/python3.6/urllib/request.py", line 248, in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:
  File "/usr/local/lib/python3.6/urllib/request.py", line 223, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/lib/python3.6/urllib/request.py", line 532, in open
    response = meth(req, response)
  File "/usr/local/lib/python3.6/urllib/request.py", line 642, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/local/lib/python3.6/urllib/request.py", line 570, in error
    return self._call_chain(*args)
  File "/usr/local/lib/python3.6/urllib/request.py", line 504, in _call_chain
    result = func(*args)
  File "/usr/local/lib/python3.6/urllib/request.py", line 650, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found
```

a 404 is something we might be able to predict, so we should catch it in ResourceUpdater.py and return a sensible error instead of a stacktrace
